src/lakefs_spec/transaction.py

```diff
@@ -132,6 +132,11 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self.fs._intrans = False
         self.fs._transaction = None
 
+        if any(True for _ in self._ephemeral_branch.uncommitted()):
+            logger.warning(
+                f"Transaction closed with uncommitted changes. {'Objects added but not committed are lost.' if self.delete != 'never' else ''}"
+            )
```

Comment:
```suggestion
        if any(self._ephemeral_branch.uncommitted()):
            msg = f"Finished transaction {self._ephemeral_branch.id!r} contains uncommitted changes."
            if self.delete != "never":
                msg += " Objects added but not committed are lost."
            warnings.warn(msg)
```

Explanation: We always prefer a warning over a log, because that works even when the user does not set up logging. If the user needs logging, he can do `import logging; logging.capturewarnings(True)`, if he wants to catch the warnings, he can do so with a warnings filter context manager or by redirecting stderr.

Bonus: We keep a clean separation between stdout and stderr.